include/electionguard/precompute_buffers.hpp

```diff
@@ -17,129 +17,183 @@
+        operator=(const PrecomputedFakeDisjuctiveCommitments &other);
+        PrecomputedFakeDisjuctiveCommitments &
+        operator=(PrecomputedFakeDisjuctiveCommitments &&other);
+
+        std::unique_ptr<PrecomputedFakeDisjuctiveCommitments> clone();
 
-        ElementModQ *get_exp1() const { return exp1.get(); }
-        ElementModQ *get_exp2() const { return exp2.get(); }
-        ElementModP *get_g_to_exp1() const { return g_to_exp1.get(); }
-        ElementModP *get_g_to_exp2_mult_by_pubkey_to_exp1() const
-        {
-            return g_to_exp2_mult_by_pubkey_to_exp1.get();
-        }
+        /// <summary>
+        /// The first secret value (either 𝑢0 or 𝑢1 in the spec)
+        /// </summary>
+        ElementModQ *getSecret1() const { return secret1.get(); }
+
+        /// <summary>
+        /// The second secret value (w in the spec)
+        /// </summary>
+        ElementModQ *getSecret2() const { return secret2.get(); }
+
+        /// <summary>
+        /// The pad applied to the ciphertext message (either a0 or a1 in the spec)
+        /// </summary>
+        ElementModP *getPad() const { return pad.get(); }
+
+        /// <summary>
+        /// The fake data proving an encryption of zero (b1 in the spec)
+        /// Use this value when proving an encryption of zero.
+        /// </summary>
+        ElementModP *getDataZero() const { return dataZero.get(); }
+
+        /// <summary>
+        /// The pad applied to the ciphertext message (b0 in the spec)
+        /// Use this value when proving an encryption of one.
+        /// </summary>
+        ElementModP *getDataOne() const { return dataOne.get(); }
 
       protected:
-        void generateQuadruple(const ElementModP &publicKey);
+        void generate(const ElementModP &publicKey);
+
+      private:
+        std::unique_ptr<ElementModQ> secret1;
+        std::unique_ptr<ElementModQ> secret2;
+        std::unique_ptr<ElementModP> pad;
+        std::unique_ptr<ElementModP> dataZero;
+        std::unique_ptr<ElementModP> dataOne;
     };
 
     /// <summary>
-    /// This object holds the two Triples and a Quadruple of precomputed
-    /// values that are used to speed up encryption of a selection.
-    /// Since the values are precomputed it removes all the exponentiations
-    /// from the ElGamal encryption of the selection as well as the
-    /// computation of the Chaum Pedersen proof.
+    /// The PrecomputedSelection is a set of precomputed values that are used
+    /// to speed up encryption of a selection. It removes most the exponentiations
+    /// from the ElGamal encryption of the selection as well as the computation
+    /// of the Chaum Pedersen proof.
+    ///
+    /// This object holds a Precomputed Encryption for the selection, a Precomputed
+    /// Encryption for the proof, and a Precomputed Fake Disjunctive Commitment for
+    /// the proof.
     /// </summary>
-    class EG_API TwoTriplesAndAQuadruple
+    class EG_API PrecomputedSelection
```

Comment:
here's where it comes together. currently there are 2 caches. one for just "Triples" or PrecomputedEncryptions and another for "everything needed for a selection" or PrecomputedSelection objects.